### ISSUE TITLE
Negative Numbers which client receives dont get parsed correctly

### ISSUE THREAD

**Issue by Wenish** [gamestdio/colyseus-unity3d#36]
If i set a position of a Player in the server state to this:

```js
currentPosition: {
  x: -30,
  y: 2.02,
  z: 0
}
```

on the client its this:
```csharp
currentPosition: {
  x: 2,
  y: 2.02,
  z: 0
}
```

Client Side Code which prints out x wrong.
in debug u see its in DataChange already false
```csharp
void OnPlayerChange (DataChange change)
{
	Debug.Log ("OnPlayerChange");
	Debug.Log (change.operation);
	Debug.Log (change.path["id"]);
	Debug.Log (change.value);

	var data = (IndexedDictionary<string, object>) change.value;

	var currentPosition = (IndexedDictionary<string, object>) data["currentPosition"];

	var x = Convert.ToSingle (currentPosition ["x"]); //prints out 2 but should be -30
	var y = Convert.ToSingle (currentPosition ["y"]);
	var z = Convert.ToSingle (currentPosition ["z"]);
}
```

Modified Version of the Unity client example which set the spawn of the cube to this
![addplayer](https://user-images.githubusercontent.com/18367963/37965546-b988fc1c-31c5-11e8-9dd3-42d00fdb50dc.png)

modified client so he sets also the Z:
![grabvalue](https://user-images.githubusercontent.com/18367963/37965570-ca9eb8fc-31c5-11e8-8eae-dc238b81b72e.png)

actual spawn: (ignore X because its already moved by script)
![spawn](https://user-images.githubusercontent.com/18367963/37965590-dae30100-31c5-11e8-83c7-5fdb4b5ef034.png)

[colyseus-unity3d-master-wrong-parsing.zip](https://github.com/gamestdio/colyseus-unity3d/files/1851806/colyseus-unity3d-master-wrong-parsing.zip)

**Comment by endel** [comment]
Thanks a lot, @Wenish! I've dug down and this has something to do with the msgpack decoder. Created an issue on the owner's repository, let's see if he can help us. https://github.com/deniszykov/msgpack-unity3d/issues/18

**Comment by endel** [comment]
Hi @Wenish, good news! @deniszykov managed to fix this on version `2.3.2` of the msgpack library! I've updated his plugin here. Can you download the latest version and try again? Cheers!
